extension/src/popup/Router.tsx

```diff
@@ -251,9 +251,9 @@ export const Router = () => {
         <PublicKeyRoute path={ROUTES.viewPublicKey}>
           <ViewPublicKey />
         </PublicKeyRoute>
-        <PrivateKeyRoute path={ROUTES.signTransaction}>
+        <PublicKeyRoute path={ROUTES.signTransaction}>
           <SignTransaction />
-        </PrivateKeyRoute>
+        </PublicKeyRoute>
```

Comment:
we're handling private key fetching using the VerifyAccount view so we don't need to auto check for private key here